biz.aQute.remote/src/aQute/remote/agent/RedirectOutput.java

```diff
@@ -85,6 +69,31 @@ public void write(byte b[], int off, int len) {
 
 	public void flush() {
 		super.flush();
+
+		for (AgentServer agent : agents) {
+			if (agent.quit)
+				continue;
+
+			try {
+				if (err)
+					agent.getSupervisor().stderr(sb.toString());
+				else
+					agent.getSupervisor().stdout(sb.toString());
+			}
+			catch (InterruptedException ie) {
+				return;
+			}
+			catch (Exception ie) {
+				try {
+					agent.close();
+				}
+				catch (IOException e) {
+					// e.printStackTrace();
+				}
+			}
+		}
+
+		sb = new StringBuilder();
```

Comment:
Perhaps you should convert the StringBuffer to a local String (and use in the loop) and reset sb before the for loop?